src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -114,24 +117,24 @@ public HikariPool(final HikariConfig config)
       ThreadFactory threadFactory = config.getThreadFactory();
 
       final int maxPoolSize = config.getMaximumPoolSize();
-      this.addConnectionExecutor = createThreadPoolExecutor(maxPoolSize, poolName + ":connection-adder", threadFactory, new CustomDiscardPolicy());
+      this.addConnectionExecutor = createCreatorThreadPoolExecutor(
+         Math.min(Runtime.getRuntime().availableProcessors() * 2, config.getMinimumIdle()),
```

Comment:
Also, core pool size will have to be the same as max pool size, so that executor doesn't hold the task in the queue and executes immediately